src/components/body/BodyController.js

```diff
@@ -137,13 +155,17 @@ export default class BodyController {
         }
 
         if (this.options.paging.externalPaging) {
+          // We're using external paging
           let idxs = this.getFirstLastIndexes(),
               idx = idxs.first;
 
           this.tempRows.splice(0, this.tempRows.length);
           while (idx < idxs.last) {
             this.tempRows.push(rows[idx++])
           }
+        } else if (this.options.paging.size) {
```

Comment:
It looks like this is nested in a this.options.paging.size already, which would prevent the next else statement from ever firing. Looks safe to remove the if condition and the else below it, but you may know more about this area than me now :)

That being said - do we know the relation of the externalPaging logic to all of that paging that happens in virtual paging?